onnx/test/model_inference_test.py

```diff
@@ -132,6 +133,53 @@ def test_mi_function_multiple_calls(self):
         """
         self._check(model, onnx.TensorProto.INT32, onnx.TensorProto.INT64)
 
+    def _check_shape(self, model_text: str, *expected: typing.Sequence[int]):
+        """Check that the model inference infers the expected shapes for outputs.
+        Restricted to the simple case of tensor type outputs with completely
+        known shapes.
+        """
+        model = onnx.parser.parse_model(model_text)
+        inferred = onnx.shape_inference.infer_shapes(model, True, True, True)
+        outputs = inferred.graph.output
+        for (output, expected_shape) in zip(outputs, expected):
```

Comment:
Assert equal length and non empty?

Reply:
Not necessary, I feel. That wouldn't be testing the inference-code.

Reply:
Just wanted to make sure the loop is run at least once. NBD